code/datums/action.dm

```diff
@@ -708,19 +726,60 @@
 	button.maptext_height = 12
 	return button
 
+/datum/action/cooldown/Destroy()
+	if(!isnull(sequence_actions))
+		sequence_actions.Cut()
+	QDEL_LIST(initialized_actions)
+	return ..()
+
+/datum/action/cooldown/Grant(mob/M)
```

Comment:
Changing `M` to a better var name (`granted_to`, for example) would be nice

(Only need to do it with this override, don't need to do it for all `Grant()`s)